configuretools.cmake

```diff
@@ -0,0 +1,75 @@
+# Compiler tools only have the major and minor version in their name
+if(CMAKE_CXX_COMPILER_VERSION MATCHES "^([0-9]+).([0-9]+)")
```

Comment:
Does it always contain a version? Even if the clang executable is named clang++ and there is no clang++-x.y?

Reply:
Yes. CMake determines the version from means other than the name. It pulls it from `clang --version` or `gcc --version` or other related means.